### ISSUE TITLE
Set bonuses - set.yml [Suggestion]

### ISSUE THREAD

**Issue by UltraFaceguy** [deathmarine/DiabloDrops#15]
Basically you can define sets in a set.yml file. Format would be as follows.

Faceguy's:                                                                   # Name of set
...Type: prefix                                                               # Prefix or suffix
...Rank: Magical                                                          # Or just color, if it makes things easier
...Tier: Gold                                                                  # Gold Diamond etc, or ANY
...Bonus:                                                                      # Set Bonus
......-1 Speed
......+1 Lightning
......- 1 AnyGivenLoreEnchantmentThatCanBeUsedInLore.txt

Basically it would work like this: Upon item creation, a check would be made on the set.yml for a matching name (Faceguy's). It would then check the type. If 'Faceguy's' is the prefix, it continues on to check the rank (or color) of the item, and optionally the tier of the item. If it passes all of this, text is added saying 

"Faceguy's set - Wearing all pieces of this set grants the following enhancements.
-1 Speed
+1 Lightning
-1 AnyGivenLoreEnchantmentThatCanBeUsedInLore.txt"

This would apply once, and not for each item.

So yeah that's my idea. Let me know what you think, and feel free to alter it in any way shape or form!
